charts/agent/tests/api_endpoint_region_test.yaml

```diff
@@ -0,0 +1,66 @@
+suite: Test agent configmap collector value set by api_endpoint for all regions (us1,us2,us3,us4,eu1,au1)
+templates:
+  - templates/configmap.yaml
+tests:
+  - it: Checking default value no region specified (us1)
+    asserts:
+      - isKind:
+          of: ConfigMap
+      - matchRegex:
+          path: data.dragent\.yaml
+          pattern: .*collector\.sysdigcloud\.com.*
+  - it: Checking region 'us2'
+    set:
+      global: 
+        sysdig:
+          region: us2
+    asserts:
+      - isKind:
+          of: ConfigMap
+      - matchRegex:
+          path: data.dragent\.yaml
+          pattern: .*ingest\-us2\.app\.sysdig\.com.*
+  - it: Checking region 'us3'
+    set:
+      global: 
+        sysdig:
+          region: us3
+    asserts:
+      - isKind:
+          of: ConfigMap
+      - matchRegex:
+          path: data.dragent\.yaml
+          pattern: .*ingest\.us3\.app\.sysdig\.com.*
+  - it: Checking region 'us4'
+    set:
+      global: 
+        sysdig:
+          region: us4
+    asserts:
+      - isKind:
+          of: ConfigMap
+      - matchRegex:
+          path: data.dragent\.yaml
+          pattern: .*ingest\.us4\.sysdig\.com.*
+  - it: Checking region 'eu1'
+    set:
+      global: 
+        sysdig:
+          region: eu1
+    asserts:
+        - isKind:
+            of: ConfigMap
+        - matchRegex:
+            path: data.dragent\.yaml
+            pattern: .*ingest\-eu1\.app\.sysdig\.com.*
+  - it: Checking region 'au1'
+    set:
+      global: 
+        sysdig:
+          region: au1
+    asserts:
+      - isKind:
+          of: ConfigMap
+      - matchRegex:
+          path: data.dragent\.yaml
+          pattern: .*ingest\.au1\.sysdig\.com.*
```

Comment:
just for my own curiosity, why do we have `asserts.matchRegex.path` here set to `data.dragent\.yaml`, but in the node analyzer test we use `asserts.equal.path` and `asserts.equal.value`? The method used in the node analyzer check seems to be more specific and clear about what it's checking. Should they be made consistent?

Reply:
Oh yeah, this one was pretty annoying to deal with but the value inside data.dragent.yaml where is as follows
```
data:
  dragent.yaml: |
    new_k8s: true
    k8s_cluster_name: yash-k8s
    security:
      enabled: true
    commandlines_capture:
      enabled: true
    memdump:
      enabled: true

    collector: collector.sysdigcloud.com
    k8s_coldstart:
      max_parallel_cold_starts: 1
```
since the whole value is a block of values because of the "|" present above, matchRegex was a better way of verifying it.

Reply:
ah, gotcha. sounds good! thanks for explaining that.